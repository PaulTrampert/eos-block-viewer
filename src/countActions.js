export default (block) => {
  let actions = 0;
  for (let transaction of block.transactions) {
    if (transaction.trx.transaction) {
      actions += transaction.trx.transaction.actions.length;
    }
  }
  return actions;
}