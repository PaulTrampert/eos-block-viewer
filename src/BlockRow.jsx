import React from 'react';
import PropTypes from 'prop-types';
import countActions from './countActions';

class BlockRow extends React.Component {
  render = () => {
    let {block} = this.props;

    return (
      <tr>
        <td>{block.id}</td>
        <td>{new Date(block.timestamp).toLocaleString()}</td>
        <td>{countActions(block)}</td>
      </tr>
    );
  }
}

BlockRow.propTypes = {
  block: PropTypes.object
};

export default BlockRow;