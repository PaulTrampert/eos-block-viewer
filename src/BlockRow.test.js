import React from 'react';
import {shallow} from 'enzyme';
import BlockRow from './BlockRow';
import countActions from './countActions';

jest.mock('./countActions');

describe('BlockRow', () => {
  let block;
  let subject;

  beforeEach(() => {
    block = {
      id: '1',
      timestamp: '2020-02-05T03:46:32.000',
      transactions: []
    };

    countActions.mockReturnValue(0);

    subject = shallow(<BlockRow block={block} />);
  });

  describe('render', () => {
    it('calls countActions to count the actions', () => {
      expect(countActions).toHaveBeenCalledWith(block);
    });

    it('renders the block data in a table row', () => {
      expect(subject).toMatchSnapshot();
    });
  });
});