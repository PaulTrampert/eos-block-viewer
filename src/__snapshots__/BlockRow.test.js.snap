// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockRow render renders the block data in a table row 1`] = `
<tr>
  <td>
    1
  </td>
  <td>
    2/5/2020, 3:46:32 AM
  </td>
  <td>
    0
  </td>
</tr>
`;
